org.eclipse.xtend.core/xtend-gen/org/eclipse/xtend/core/macro/declaration/JvmExecutableDeclarationImpl.java

```diff
@@ -140,6 +140,22 @@ public MutableParameterDeclaration addParameter(final String name, final TypeRef
     return ((MutableParameterDeclaration) _parameterDeclaration);
   }
 
+  public MutableParameterDeclaration addParameter(final int index, final String name, final TypeReference type) {
```

Comment:
Would it make sense to delegate from `addParameter(String, TypeReference)` to this new method?